sklearn/pipeline.py

```diff
@@ -981,6 +998,11 @@ def _update_transformer_list(self, transformers):
                                      else next(transformers))
                                     for name, old in self.transformer_list]
 
+    def _more_tags(self):
+        # The FeatureUnion can handle NaNs if all the steps can.
+        return {'allow_nan': np.all([s[1]._get_tags()['allow_nan']
+                                     for s in self.transformer_list])}
```

Comment:
Perhaps remove the list comprehension: use a generator. It's just one more character of bracketing for the reader to decide at the moment